packages/aws-event-stream/src/aws_event_stream/_private/serializers.py

```diff
@@ -36,16 +36,23 @@
 _DEFAULT_BLOB_CONTENT_TYPE = "application/octet-stream"
 
 
-type Signer = Callable[[EventMessage], EventMessage]
-"""A function that takes an event message and signs it, and returns it signed."""
+class EventSigner(Protocol):
+    """A signer to manage credentials and EventMessages for an Event Stream lifecyle."""
+
+    def sign_event(
+        self,
+        *,
+        event_message: EventMessage,
+        event_encoder_cls: type[EventHeaderEncoder],
```

Comment:
This shouldn't be needed anymore right? It'll just get encoded later

Reply:
I think its necessary to encode the new headers while building the string to sign.

Reply:
Yeah this was before my Grand Revelation